internal/integratedservices/servicerouter.go

```diff
@@ -58,35 +58,64 @@ func (s serviceRouter) List(ctx context.Context, clusterID uint) ([]IntegratedSe
 // Details retrieves the service from the service v1 if not found retrieves it from v2
 // Note: an Integrated Service can only be managed by one of the service versions
 func (s serviceRouter) Details(ctx context.Context, clusterID uint, serviceName string) (IntegratedService, error) {
-	if detailsV1, err := s.serviceV1.Details(ctx, clusterID, serviceName); err == nil {
-		// return the legacy integrated service details
-		return detailsV1, nil
-	} else if !IsIntegratedServiceNotFoundError(err) {
-		return IntegratedService{}, errors.Wrapf(err, "failed to retrieve legacy integrated service details")
-	} // ignore the not found error, proceed to the new implementation
+	var (
+		// cache variables to spare redundant external calls
+		detailsV1, detailsV2 IntegratedService
+		errV1, errV2         error
+	)
+
+	if detailsV1, errV1 = s.serviceV1.Details(ctx, clusterID, serviceName); errV1 == nil {
+		if detailsV1.Status != IntegratedServiceStatusInactive {
+			// return the legacy integrated service details
+			return detailsV1, nil
+		}
+	} else if !IsUnknownIntegratedServiceError(errV1) {
+		// ignore the unknown service error, proceed to the new implementation
```

Comment:
this comment is confusing here since we are processing the errors which are not the Uknown ones instead of ignoring anything